src/test/components/WindowTitle.test.js

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React from 'react';
+import { Provider } from 'react-redux';
+import { render } from 'react-testing-library';
+
+import WindowTitle from '../../components/shared/WindowTitle';
+import {
+  getEmptyProfile,
+  getEmptyThread,
+} from '../../profile-logic/data-structures';
+import { storeWithProfile } from '../fixtures/stores';
+import { changeProfileName } from '../../actions/profile-view';
+
+describe('WindowTitle', () => {
+  it('shows basic window title', () => {
+    const profile = getEmptyProfile();
+    profile.threads.push(getEmptyThread());
+    const store = storeWithProfile(profile);
+    render(
+      <Provider store={store}>
+        <WindowTitle />
+      </Provider>
+    );
+
+    expect(document.title).toBe(
+      'Firefox - 1/1/1970, 12:00:00 AM UTC - Firefox profiler'
+    );
+  });
+
+  it('shows platform details in the window title if it is available', () => {
+    const profile = getEmptyProfile();
+    profile.threads.push(getEmptyThread());
+    profile.meta.oscpu = 'Intel Mac OS X 10.14';
+    const store = storeWithProfile(profile);
+    render(
+      <Provider store={store}>
+        <WindowTitle />
+      </Provider>
+    );
+
+    expect(document.title).toBe(
+      'Firefox - Intel Mac OS X 10.14 - 1/1/1970, 12:00:00 AM UTC - Firefox profiler'
+    );
+  });
+
+  it('shows profile name in the window title if it is available', () => {
+    const profile = getEmptyProfile();
+    profile.threads.push(getEmptyThread());
+    profile.meta.oscpu = 'Intel Mac OS X 10.14';
+    const store = storeWithProfile(profile);
+    store.dispatch(changeProfileName('good profile'));
+    render(
+      <Provider store={store}>
+        <WindowTitle />
+      </Provider>
+    );
+
+    expect(document.title).toBe(
+      'good profile - Firefox - Intel Mac OS X 10.14 - 1/1/1970, 12:00:00 AM UTC - Firefox profiler'
+    );
+  });
```

Comment:
Can you add another check in this last test: can you dispatch another profile name change, and check the document.title again -- without rendering again ?  The goal is to check whether updates work fine.

Then I think we'd be good to merge this :)